addons/binding/org.openhab.binding.spotify/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,366 @@
+		</config-description>
+	</thing-type>
+
+	<!-- Channel Types -->
+	<channel-type id="accessToken" advanced="true">
+		<item-type>String</item-type>
+		<label>Access Token</label>
+		<description>This channel provides the access token used to authorize calls to the Spotify Web API. It can be used from rules or client-side scripting make call to Web API.</description>
+	</channel-type>
+	<channel-type id="activeDeviceId" advanced="true">
+		<item-type>String</item-type>
+		<label>Active Device Id</label>
+		<description>Channel to get/set active device id</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="activeDeviceName">
+		<item-type>String</item-type>
+		<label>Active Device Name</label>
+		<description>Channel reports active device name</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="activeDeviceType" advanced="true">
+		<item-type>String</item-type>
+		<label>Active Device Type</label>
+		<description>Channel reports active device type</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="activeDeviceVolume">
+		<item-type>Dimmer</item-type>
+		<label>Active Device Volume</label>
+		<description>Channel get/set active device volume</description>
+		<category>SoundVolume</category>
+	</channel-type>
+	<channel-type id="activeDeviceShuffle">
+		<item-type>Switch</item-type>
+		<label>Active Device Shuffle</label>
+		<description>Channel to get/set shuffle for active device</description>
+	</channel-type>
+
+	<channel-type id="playlist">
+		<item-type>String</item-type>
+		<label>Playlist</label>
+		<description>Channel with the users playlists. Select playlist to play</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="trackPlay" advanced="true">
+		<item-type>String</item-type>
+		<label>Track to Play</label>
+		<description>Channel to assign track/album/playlist to play. Accepts Spotify Uri:s and Hrefs.</description>
+	</channel-type>
+	<channel-type id="trackPlayer">
+		<item-type>Player</item-type>
+		<label>Player Control</label>
+		<description>Channel for player control</description>
+		<category>Player</category>
+	</channel-type>
+	<channel-type id="trackRepeat">
+		<item-type>String</item-type>
+		<label>Repeat Mode</label>
+		<description>Track will repeat the current track. Context will repeat the current context. Off will turn repeat off.</description>
+		<state>
+			<options>
+				<option value="track">Track</option>
+				<option value="context">Context</option>
+				<option value="off">Off</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="currentlyPlayedTrackId" advanced="true">
+		<item-type>String</item-type>
+		<label>Track Id</label>
+		<description>Channel reports track id currently being played.</description>
+		<state readOnly="true" />
+	</channel-type>
+	<channel-type id="currentlyPlayedTrackHref" advanced="true">
+		<item-type>String</item-type>
+		<label>Track href</label>
```

Comment:
This label is not very descriptive, it also does not follow the capitalised-words syntax.